### PR TITLE
feat: Update Makefile, consume github-actions@v3.0.0

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Relates to https://github.com/kubewarden/kubewarden-controller/issues/389

## Test

Not tested.

### CODE REVIEW

**Review by jvanz — APPROVED** [review]

**Comment by flavio** [comment]
> * a `0.0.1` git tag in this repo

Maybe we should start to have tags on this repository. It would make easier to understand what is going on. I just wonder how this would impact on people who create new repositories based on this template. I doubt the tags would propagate to their repos...

> * none, deactivate check artifacthub

Yes, that's the other fix I would do. Just update the settings of this repo to ignore these failures. I hope/doubt this would be propagated to "forked" repos...

File: Makefile [hunk 1]
```diff
@@ -1,20 +1,27 @@
 SOURCE_FILES := $(shell find . -type f -name '*.go')
-VERSION := $(shell git describe --exact-match --tags $(git log -n1 --pretty='%h') | cut -c2-)
+VERSION := $(shell git describe | cut -c2-)
```
Comment (flavio) [inline Makefile:2]: Shouldn't this be:

```
VERSION ?= $(shell git describe | cut -c2-)
```

Comment (viccuad) [reply Makefile]: true, done in https://github.com/kubewarden/go-policy-template/pull/39/commits/bc65b8ad50196dd5d81408bb09cb859c65e31b8b.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -3,4 +3,4 @@ name: Continuous integration
 jobs:
   test:
     name: run tests and linters
-    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v2
+    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v3.0.0
```
Comment (jvanz) [inline .github/workflows/test.yml:6]:
```suggestion
    uses: kubewarden/github-actions/.github/workflows/reusable-test-policy-go.yml@v3.0.4
```

**Review by jvanz — CHANGES_REQUESTED** [review]

**Comment by jvanz** [comment]
> Maybe we should start to have tags on this repository. It would make easier to understand what is going on. I just wonder how this would impact on people who create new repositories based on this template. I doubt the tags would propagate to their repos.

Yes, I think is a good idea. Considering the github docs:
```
Creating a repository from a template is similar to forking a repository, but there are important differences:

    A new fork includes the entire commit history of the parent repository, while a repository created from a template starts with a single commit.
```

I think the tags will not affect the users creating the repos from the template. If the "forked" repository will have a single commit, there is no reason to import the tags as well. The only consideration is to how we add branches in the template repository. Because user may import them into their repositories:

```
After you make your repository a template, anyone with access to the repository can generate a new repository with the same directory structure and files as your default branch. They can also choose to include all the other branches in your repository. Branches created from a template have unrelated histories, so you cannot create pull requests or merge between the branches. 
```

https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-template-repository
https://docs.github.com/en/repositories/creating-and-managing-repositories/creating-a-repository-from-a-template

**Comment by viccuad** [comment]
So, as a test,
- Created a `v0.0.1-unreleased` tag in this repo: https://github.com/kubewarden/go-policy-template/tree/v0.0.1-unreleased
- Created a fork from the template: https://github.com/viccuad/my-go-template-fork

My fork doesn't get the tag by default, i need to `git fetch && git push -t`. We could add a check to the Makefile: "no tags detected, please make your first or do `git fetch && git push -t`".

**Comment by flavio** [comment]
Good! let's get this PR merged and then we can start tagging this repo as well

**Comment by viccuad** [comment]
Merging, leaving the tag in the repo.

**Pull request merged**